xaitk_saliency/impls/gen_detector_prop_sal/drise_scoring.py

```diff
@@ -108,9 +108,9 @@ def generate(
                       ref_dets[:, :4]).reshape(n_masks, n_props, n_dets)
 
         # Compute similarity of class probabilities
-        s2 = cdist(perturbed_dets[:, :, 5:].reshape(n_masks * n_props, -1),
-                   ref_dets[:, 5:],
-                   metric=self.proximity_metric).reshape(n_masks, n_props, n_dets)
+        s2 = 1 - cdist(perturbed_dets[:, :, 5:].reshape(n_masks * n_props, -1),
+                       ref_dets[:, 5:],
+                       metric=self.proximity_metric).reshape(n_masks, n_props, n_dets)
```

Comment:
One issue here is that this change to the `1 - cdist(...)` operation may no longer be valid for other non-cosine proxity metrics, e.g. Euclidean distance. I'm not sure how to best address this, may be with a warning in the `__init__` block?

Reply:
If we only know D-RISE to "work" with cosine similarity, should we then just take out the option to configure to other things?

Reply:
This is kind of a philosophical question- we also use the `cdist` operation in our `SimilarityScoring` implementation, and from what I can tell, the original implementation mainly tested it with Euclidean distance, yet we currently make available the whole gamut of distance metrics, many of which may not be valid for that implementation.

Reply:
It could be the case that we over-engineered that impl as well, however it seems that works on distance (metric?) functions, which all the cdist options are I thought. It's also OK if we "fix" this impl while the other one is still under question.

Reply:
I agree that, if the impl only really "works" w/ a specific distance metric, than perhaps it's better not to parametrize this